providers/process/abstractProcessor.js

```diff
@@ -0,0 +1,138 @@
+const { promisify } = require('util')
+const nodeDir = promisify(require('node-dir').files)
+const { trimAllParents } = require('../../lib/utils')
+
+class AbstractProcessor extends BaseHandler {
+  get schemaVersion() {
+    return '0.0.0'
+  }
+
+  _computeToken(content) {
+    return shajs('sha256')
+      .update(content)
+      .digest('hex')
+  }
+
+  /**
+   * Attach the files at the given `paths` (relative to the identified `location`) to the document
+   *
+   * @param {Object} document - The document to host the attachments
+   * @param {[string]} files - Relative paths to the attachment files
+   * @param {string} location - Root filesystem path that hosts the files to be attached
+   */
+  attachFiles(document, files, location = '') {
+    if (!files || !files.length) return
+    if (!document._attachments) Object.defineProperty(document, '_attachments', { value: [], enumerable: false })
+    document.attachments = document.attachments || []
+    files.forEach(file => {
+      const fullPath = path.join(location, file)
+      const attachment = fs.readFileSync(fullPath, 'utf8')
+      const token = this._computeToken(attachment)
+      // Stash the actual content on a hidden prop on the document and note the file in the list of attachments
+      document._attachments.push({ path: file, token, attachment })
+      document.attachments.push({ path: file, token })
+    })
+  }
+
+  /**
+   * Get the list of all files in the filesytem under the given location
+   * @param {String} location - file system location to search
+   * @returns {String[]} - full file system paths of all files found
+   */
+  getFiles(location) {
+    return nodeDir(location)
+  }
+
+  /**
+   * Find all the files that are thought to be interesting relative to location.
+   * @param {String} location - file system location to search
+   * @returns {String[]} - location-relative paths of interesting files found. Note that all paths
+   * are normalized to use '/' as the separator
+   */
+  async filterFiles(location) {
+    const fullList = await this.getFiles(location)
+    const exclusions = ['.git']
+    const filteredList = fullList.filter(file => {
+      if (!file) return false
+      const segments = file.split(/[\\/]/g)
+      return !intersection(segments, exclusions).length
+    })
+    return trimAllParents(filteredList, location)
+  }
+
+  shouldFetch() {
+    return true
+  }
+
+  canHandle() {
+    return false
+  }
+
+  shouldProcess(request) {
+    return request.policy.shouldProcess(request, this.schemaVersion)
+  }
+
+  shouldTraverse(request) {
+    return request.policy.shouldTraverse(request)
+  }
+
+  isProcessing(request) {
+    return request.processMode === 'process'
+  }
+
+  handle(request) {
+    set(request, 'document._metadata.version', this.schemaVersion || 1)
+    const spec = this.toSpec(request)
+    this.addBasicToolLinks(request, spec)
+  }
+
+  clone(document) {
+    return pick(document, ['_metadata'])
```

Comment:
we are dropping attachments here. 

```suggestion
    return pick(document, ['_metadata', '_attachments', 'attachments'])
```

Reply:
however, when I did this, it seems to have lost it's `enumerable: false` attribute and is showing up in the harvested document :(

Reply:
resolved in 6f78032f7b5de3bd38720e3984f51b4d1dc0dab4